src/coreclr/tools/Common/TypeSystem/Interop/IL/Marshaller.Aot.cs

```diff
@@ -925,4 +927,29 @@ protected override void TransformManagedToNative(ILCodeStream codeStream)
             throw new NotSupportedException();
         }
     }
+
+    class BSTRStringMarshaller : Marshaller
+    {
```

Comment:
This also needs to override `CleanupRequired` and provide implementation of `EmitElementCleanup`